### DISCUSSION TITLE
Getting "Argument can only be defined once" errors after upgrades

### DISCUSSION THREAD

**Discussion by jlleblanc** [mirumee/ariadne#1118] [Q&A] [answered]
Hello everyone! We have been successfully using Ariadne in Django for a while now and are in the middle of doing some upgrades. Previously, we were on Python 3.8 with the following relevant dependencies:

```
ariadne==0.9.0
gql==3.0.0a5
graphql-core==3.1.1
typing-extensions==3.7.4.1
```

When I tried upgrading to Python 3.9, I did a bit of dependency juggling and ended up updating all of them:

```
ariadne==0.20.1
ariadne-django==0.3.0
gql==3.4.1
graphql-core==3.2.3
typing-extensions==4.7.1
```

(Note that `ariadne-django` is an addition here, we were not using it before.)

After doing these upgrades, `make_executable_schema` is tripping up on some queries. Here is an example:

```
Argument 'Query.findUser(after:)' can only be defined once.
```

We are taking several `.graphql` files, concatenating them together, then passing them into `make_executable_schema`. I have confirmed that this variable we are passing does not contain any duplicated query definitions (i.e., there is only one `findUser`).

The error is happening in `graphql-core` where `validate_sdl` is called. But further up the stack, Ariadne is calling `build_ast_schema` with one argument. When we dug into `build_ast_schema`, we discovered two keyword arguments that default to `False`: `assume_valid` and `assume_valid_sdl`.

I forked `graphql-core` to [set these defaults](https://github.com/graphql-python/graphql-core/commit/27bc2415fe46d60ca889cc83014879d383020a9b) to `True`. After doing that and using the fork in our project, the schema loaded with no errors.

I could proceed with that fork. I could also alternatively fork Ariadne to set everything to assume valid SDL. But I would rather not do that for maintainability reasons. I am also wondering why our type definitions are considered to be invalid.

Has anyone else run into this?

**Comment by rafalp** [comment]
Hello, can you please share the SDL for `findUser` field on `Query` type?

**Reply by jlleblanc** [reply]
Sure thing! Let me know if there is anything else that would be helpful to share.

```gql
extend type Query {
  me: User
  findUser(
    first: Int
    after: String
    last: Int
    after: String
    filter: UserFilter
  ): UserConnection
}

type UserConnection {
  pageInfo: PageInfo!
  edges: [UserEdge]!
  total: Int
}

type UserEdge {
  node: User!
  cursor: String!
}
```

**Reply by rafalp** [reply, marked as answer]
As error says, you have `after: String` argument defined twice on `findUser` field on your query. This is invalid and you should delete one of those.

**Reply by jlleblanc** [reply]
I see now, thank you! This is my first major GraphQL project and I misread the error.
